src/javascript/app_2/App/Components/Form/input_field.jsx

```diff
@@ -32,7 +32,7 @@ const InputField = ({
             const is_empty = !e.target.value || e.target.value === '';
             const signed_regex = is_signed ? '[\\+-]?' : '';
 
-            const is_number = new RegExp(`^${signed_regex}(\\d*)?${is_float ? '(\\.\\d+)?' : ''}(?<=\\d)(?<!-0)$`)
+            const is_number = new RegExp(`^${signed_regex}(\\d*)?${is_float ? '(\\.\\d+)?' : ''}(\\d*)?(?!([-0])).|^$`)
```

Comment:
@morteza-binary I didn't quite understand the first bullet point. As you know, look-behind is not supported in all major browsers, so we have to imitate that behavior with other rules. 
Are we validating numbers only?

Reply:
@morteza-binary Would you check these examples and the new regex that handles all that as well? (this one doesn't use lookbehind and supports safari too). 
```javascript
const regex = /(?!^\-0$)(^[\-+]?(\d*)?(\.\d+)?$)/gm;
const str = `*3213.
.341341
e10923
asdada
-123
090)()91231
-0
-0.134
-.13451
2345.
1234
+1234
-3
-123
-12345161
12314151
+123e12346
0.2313
--123
123151511513510
00.00
123.`;
let m;

while ((m = regex.exec(str)) !== null) {
    // This is necessary to avoid infinite loops with zero-width matches
    if (m.index === regex.lastIndex) {
        regex.lastIndex++;
    }
    
    // The result can be accessed through the `m`-variable.
    m.forEach((match, groupIndex) => {
        console.log(`Found match, group ${groupIndex}: ${match}`);
    });
}
```

Reply:
@easteregg 
It seems that it supports a wide range of templates, but there is an issue in this regex that asserts `-` or `+` as a valid number.

Reply:
You are correct. I'll work on that case as well.